### ISSUE TITLE
Wrong handling of layers' groups

### ISSUE THREAD

**Issue by spono** [opengisch/qfieldsync#587]
Hi,
I think there's something going wrong with the handling of layer groups.
In the example picture below (screenshot from QGIS), the two layers included into the group "RASTER BASE" are used to create a basemap through the _Map Theme_ option.

![qfield_project](https://github.com/opengisch/qfieldsync/assets/14940295/2efe08a7-1b16-4850-aa69-8a3956e09fc8)

Once packed up and sent to the device, the layer group is retained even if the contained layers are removed (with the setting _Remove from project_).

![Screenshot_2024-06-05-10-47-12-58_2e601d870d31f598a7096b1e53961eb8](https://github.com/opengisch/qfieldsync/assets/14940295/d25817f7-825f-471b-b4c7-5865b6fd58c5)

Another weird beviour appears with the layer "schema_campionamento":
- if it is included into a group, it is exported but cannot be shown on the map (i.e. tapping on the layer and then on the _Show on map_ option, nothing happens)
- if it is NOT included into a group, it is correctly exported.

Tested on 4.6.4 and 4.9.1, on W10.

The project is a thinned version of the main one, due to other issues happening during the export (related to the creation of a basemap which includes OSM)

**Comment by suricactus** [comment]
Can you please upload the minimal project here, so we can confirm and identify the issue. Also, based on your writing I assume you used the cable export of QFieldSync, not QFieldCloud. If my assumption is correct, can you please confirm that opening the packaged version appear incorrect also in QGIS, or it is only a QField issue?

**Comment by spono** [comment]
you're right for both things: I used the cable export and opening the zipped project in QGIS appears incorrect (i.e. with an empty group named "RASTER BASE").

BTW, trying to pack up a minimal project with the same settings and -partly- data, now I get an error saying "No basemap layer selected. Check configuration"...even though basemap is set to use a style merging the two layers within the RASTER BASE group.

At this point, the idea is to do a total reinstall of QGIS but, before doing it, I ask you: shall I or you prefer to keep the system like this to find the bug?

**Comment by suricactus** [comment]
Please provide us a sample project, so we can confirm your issue. 

Two things you can check before reinstalling: 
1) create a new QGIS profile (don't delete the old "broken" one) and see if you experience the same.
2) if you are a Windows user, you can install multiple QGIS versions in parallel and see if you experience the same.

**Comment by spono** [comment]
OK, I'll do in the next days. In the meantime I attach here the minimal project for the second issue (no basemap)

[Uploading minimal_project_NoBasemap_Selected.zip…]()

**Comment by spono** [comment]
> Please provide us a sample project, so we can confirm your issue.
> 
> Two things you can check before reinstalling:
> 
>     1. create a new QGIS profile (don't delete the old "broken" one) and see if you experience the same.
> 
>     2. if you are a Windows user, you can install multiple QGIS versions in parallel and see if you experience the same.

my experience remained the same and I'm not able to create a smaller project for testing this issue...I guess it's related to the project itself.
In the meantime I opened a new issue for the other problem.
